tests/syntax/test_compiler.py

```diff
@@ -363,6 +363,233 @@ def test_require(self, options, expected_name, expected_prob):
+                        value=Call(
+                            func=Attribute(
+                                value=Name(id="_Scenic_current_behavior", ctx=Load()),
+                                attr="_invokeSubBehavior",
+                                ctx=Load(),
+                            ),
+                            args=[
+                                Name(id="self", ctx=Load()),
+                                Tuple(
+                                    elts=[Constant("foo")],
+                                    ctx=Load(),
+                                ),
+                                ast.Call(
+                                    func=ast.Name("Modifier"),
+                                    args=[
+                                        ast.Constant("for"),
+                                        Constant(3),
+                                        ast.Constant("steps"),
+                                    ],
+                                    keywords=[],
+                                ),
+                            ],
+                            keywords=[],
+                        )
+                    )
+                ),
+                checkInvariants,
+            ]:
+                self.assert_invocation_check_invariants(checkInvariants)
+            case _:
+                assert False
+
+    def test_do_until(self):
+        node, _ = compileScenicAST(DoUntil(Constant("foo"), Name("condition")))
+        match node:
+            case [
+                Expr(
+                    value=YieldFrom(
+                        value=Call(
+                            func=Attribute(
+                                value=Name(id="_Scenic_current_behavior", ctx=Load()),
+                                attr="_invokeSubBehavior",
+                                ctx=Load(),
+                            ),
+                            args=[
+                                Name(id="self", ctx=Load()),
+                                Tuple(
+                                    elts=[Constant("foo")],
+                                    ctx=Load(),
+                                ),
+                                ast.Call(
+                                    func=ast.Name("Modifier"),
+                                    args=[
+                                        ast.Constant("until"),
+                                        Lambda(body=Name("condition")),
+                                    ],
+                                    keywords=[],
+                                ),
+                            ],
+                            keywords=[],
+                        )
+                    )
+                ),
+                checkInvariants,
+            ]:
+                self.assert_invocation_check_invariants(checkInvariants)
+            case _:
+                assert False
+
+    def assert_invocation_check_invariants(self, node):
```

Comment:
This method asserts that `node` is the code for running invariant checks. Used by all other cases added in this PR.